### DISCUSSION TITLE
first party solution (ability to restrict labels in comments)

### DISCUSSION THREAD

**Discussion by ghost** [bluesky-social/atproto#1965] [General Discussion]
I would like a first party solution to being able to auto block posts with labels from appearing in the comments section - not just for me, but for anyone viewing the comments.

Situation: I post a post- "hey everyone"

commenter: posts naked photo

me- i see the comment and it has a label on it, ie. nsfw - i dont want to see or be exposed to that comment in my comment section, nor do i want anyone else in my community to be exposed to that content.

**Comment by bnewbold** [comment]
This is a pretty interesting idea!

We have discussed a few variations on this internally. A key question is "labeled by who": presumably labeled by the mod services the OP account has configured. That configuration can change over time; should the rules for the thread be static over time, or update dynamically? These aren't difficult issues to resolve, but we would want to make sure the answer is harmonious and understandable as part of the overall composible moderation narrative.

That being said, there might be a simpler near-term way to incorporate this kind of intent in to the thread-gate system sooner than later.

**Reply by ghost** [reply]
For personal choice, I would like if it could have a user configured default, that I could change over time (or even go back lock a post). 
some days i may be comfortable with various content in the comments section of posts other days not so much (when its unsolicited)

the reason i came up with this was because i had a very cringe worthy personal experience where someone did post nsfw content in my comments  that ideally wouldnt happen unless (the poster) want it to.  

thread-gate would be a good short term solution as well,

**Reply by ghost** [reply]
id also include that this could be used to prevent any media from being shown in comments not just nsfw media

**Reply by ghost** [reply]
i think this is also considered reply-gating, but i would also like to throw in a few other options
ability to have a **blocked list of words** to filter. If the comment contains one of those words then the comment is blocked from everyone viewing. (this would also help with impersonation, and people trying to scam/users 

also ability to have only users who follow you be able to comment/reply to add a layer of friction to  minimize bad actors,

**Reply by bnewbold** [reply]
another one getting some attention right now is "new-account" as either a label or just metadata about the account (which can be pulled from PLC for did:plc accounts, or some "first seen" at appview). the threadgate would be "minimum-account-age" for accounts to reply.

**Comment by beholdnec** [comment]
I remember reading a discussion earlier where devs were talking about making replies work on an approval system, where each reply has to be certified by your server before it can appear in the comment thread. Has this idea been pursued further as an option?

Allowing users to control who can appear in their replies is very important IMO. Obviously blocking harassers is important, but being able to set customizable policies like "no gifs in the replies," "no-one may reply more than 5 times," and so on would be nice.

**Comment by ghost** [comment]
Another reply gating option would be nice is if 'only users who had custom domains can reply' - https://bsky.app/profile/eva.computer/post/3klqedkjbuh2q

**Comment by austinlparker** [comment]
Medium term, perhaps there should be a new category of labels for 'intents'? I'm somewhat riffing on the semi-viral post of images that you could attach to mastodon posts like "Don't suggest OSS software" or "Don't explain this to me", but I think it'd be valuable if authors could add "positive labels" to content. 

Twitter kinda had this with hashtags; stuff like '#lazyweb' for people trying to have someone else look something up. The simplest one I can think of would be some sort of 'promotional' intent that could be used for ads.
